examples/vae_text/vae_train.py

```diff
@@ -99,12 +97,13 @@ def __init__(self,
 
         if config_model.decoder_type == "lstm":
             self.decoder = tx.modules.BasicRNNDecoder(
-                vocab_size=vocab_size,
                 input_size=(self.decoder_w_embedder.dim +
                     config_model.batch_size),
+                vocab_size=vocab_size,
+                token_embedder=self._embed_fn_rnn,
                 hparams={"rnn_cell": config_model.dec_cell_hparams})
             decoder_initial_state_size = (self.decoder.cell.hidden_size,) * 2
```

Comment:
I don't think you need this `decoder_initial_state_size` here. In the code below, whenever you used this variable, you still check for `decoder_type`. Why not just directly use what was assigned to this variable in each case?

Reply:
Yes, I'll put it into `decode`.